Data/NF/Internal.hs

```diff
@@ -17,11 +21,17 @@
 -- Stability   :  experimental
 -- Portability :  portable
 --
--- This module provides the constructor for the type 'NF', allowing
--- you to create values of type 'NF' without any runtime overhead
--- if you fulfill a proof obligation that the value is already in
--- normal form.
-module Data.NF.Internal(NF(..)) where
+-- This module is intended for internal use.
+module Data.NF.Internal(
+#if __GLASGOW_HASKELL__ >= 800
+  NF(..,NF)
+#else
+  NF(..)
+#if __GLASGOW_HASKELL__ >= 710
+  , pattern NF
```

Comment:
Earlier versions (7.8) didn't offer non-trivial bidirectional synonyms;
there's nothing to be done about that.

On Jul 22, 2016 6:13 PM, "Edward Z. Yang" <notifications@github.com> wrote:

> In Data/NF/Internal.hs
> <https://github.com/ezyang/nf/pull/6#discussion_r71953602>:
>
> > @@ -17,11 +21,17 @@
> >  -- Stability   :  experimental
> >  -- Portability :  portable
> >  --
> > --- This module provides the constructor for the type 'NF', allowing
> > --- you to create values of type 'NF' without any runtime overhead
> > --- if you fulfill a proof obligation that the value is already in
> > --- normal form.
> > -module Data.NF.Internal(NF(..)) where
> > +-- This module is intended for internal use.
> > +module Data.NF.Internal(
> > +#if __GLASGOW_HASKELL__ >= 800
> > +  NF(..,NF)
> > +#else
> > +  NF(..)
> > +#if __GLASGOW_HASKELL__ >= 710
> > +  , pattern NF
>
> Ugh, it kind of sucks that we can't offer the NF constructor on earlier
> versions of Haskell. Well, I leave it to you.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ezyang/nf/pull/6/files/db57fa061540b46fb54928e8131e84be376b7fe9#r71953602>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABzi_Q4-6LOnUK2rQtP4uhBtwrhU7OHLks5qYUB-gaJpZM4JTKLc>
> .
>